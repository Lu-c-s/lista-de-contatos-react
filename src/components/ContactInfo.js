import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Grid from '@material-ui/core/Grid';
import Divider from '@material-ui/core/Divider';

const styles = {
    card: {
      minWidth: 275,
    },
    img :{
       borderRadius: 10,
       border: '4px solid light-gray',
       height: 300,
       marginTop:50,
       textAlign: 'center'
       //marginLeft:150,

    }
  };
  
  const contactInfo = (props) => {
    const { classes } = props;
    console.log(props)
    return (
      <div>
        <Card className={classes.card}>
          <CardContent>
                <Grid container spacing={24}>
                    <Grid item xs={12} sm={6}>
                        <img src={props.contact.general.avatar} className={classes.img} alt="Avatar" />
                    </Grid>
                    <Grid item xs={12} sm={6}>
                        <h2>Contact Details</h2>
                        <Divider />
                        <h3>General Information</h3>
                        <h4>{props.contact.general.firstName + " " + props.contact.general.lastName}</h4>

                        <h3>Contact Information</h3>
                        <h4>{props.contact.contact.email}</h4>
                        <h4>{props.contact.contact.phone}</h4>

                        <h3>Job</h3>
                        <h4>{props.contact.job.title}</h4>
                        <h4>{props.contact.job.company}</h4>
                        
                        <h3>Address</h3>
                        <h4>{props.contact.address.country + " - " +  props.contact.address.city}</h4>
                        <h4>{props.contact.address.street}</h4>
                        <h4>{props.contact.address.zipCode}</h4>
                      
                         
                    </Grid>
                </Grid>   
          </CardContent>
        </Card>
      </div>
    );
  }
  
contactInfo.propTypes = {
    classes: PropTypes.object.isRequired,
  };
  
  export default withStyles(styles)(contactInfo);
