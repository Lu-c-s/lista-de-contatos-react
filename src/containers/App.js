import React, { Component } from 'react';
import './App.css';

import ContactInfo from '../components/ContactInfo';
import SideBar from '../containers/SideBar';
//import CircularProgress from '@material-ui/core/CircularProgress';

class App extends Component {
  constructor(props){
    super(props);
    
    this.state = {
      clients:[],
      showing:{
        id: 0,
        show: false,
        loading: true
      }
    }
  }
 
  //Show all contact info
  showInfoHandler = ( id ) => {
     
      if(id !== -1){
        const newShow = this.state.showing;
        newShow.id = id;
        newShow.show = true;
  
        this.setState({ showing : newShow })
      }
      
  }
  
  componentDidMount(){
    let url = '../../clients.json';
    fetch(url ,{
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json'
      }
    })
    .then(res => {
      return res.json();
    })
    .then(data => {
      data.map((contact , index) => contact.general.id = index );

      setTimeout(function() {
        this.setState({clients: data});
      }
        .bind(this)
        ,1500);
    })
    .catch(error => {
      console.log(error)
    })
  }

  render() {
    let contact = null;
    
     if( this.state.showing.show ) {
       contact = <ContactInfo contact={this.state.clients[this.state.showing.id]} />;
     }

    return(
      <div>
          <SideBar contacts={this.state.clients} clicked={this.showInfoHandler} >
              {contact}
              
          </SideBar>
      </div>
    );

  }
}

export default App;
