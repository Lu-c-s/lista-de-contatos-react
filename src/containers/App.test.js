import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import ContactInfo from '../components/ContactInfo';
import Contact from '../components/Contact';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('ContactInfo renders all contact info correctly', () => {
  const contactTest = {
    "general": {
      "firstName": "Liana",
      "lastName": "Crooks",
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg",
      "id": "0"
    },
    "job": {
      "company": "Ledner, Johnson and Predovic",
      "title": "Investor Functionality Coordinator"
    },
    "contact": {
      "email": "Gerry_Hackett77@gmail.com",
      "phone": "(895) 984-0132"
    },
    "address": {
      "street": "1520 Zemlak Cove",
      "city": "New Devon",
      "zipCode": "42586-7898",
      "country": "Guinea-Bissau"
    }
  }
  const tree = renderer.create(
    <ContactInfo contact={contactTest} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

test('Contact renders contact card correcly', () => {
  const contactTest = {
    "general": {
      "firstName": "Liana",
      "lastName": "Crooks",
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg",
      "id": "0"
    },
    "job": {
      "company": "Ledner, Johnson and Predovic",
      "title": "Investor Functionality Coordinator"
    },
    "contact": {
      "email": "Gerry_Hackett77@gmail.com",
      "phone": "(895) 984-0132"
    },
    "address": {
      "street": "1520 Zemlak Cove",
      "city": "New Devon",
      "zipCode": "42586-7898",
      "country": "Guinea-Bissau"
    }
  }
  const tree = renderer.create(
    <Contact
        key={contactTest.general.id}
        id={contactTest.general.id}
        clicked={() => props.clicked( contactTest.general.id )}
        firstName={contactTest.general.firstName}
        lastName={contactTest.general.lastName}
        avatar={contactTest.general.avatar}
        company={contactTest.job.company} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
