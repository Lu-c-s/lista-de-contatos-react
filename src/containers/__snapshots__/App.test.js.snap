// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders contact card correcly 1`] = `
<div
  className="contact-root-284"
  onClick={[Function]}
>
  <div
    className="MuiButtonBase-root-295 MuiListItem-root-285 MuiListItem-default-288 MuiListItem-gutters-292 MuiListItem-button-293"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <div
      className="MuiAvatar-root-298"
    >
      <img
        alt={undefined}
        className="MuiAvatar-img-300"
        sizes={undefined}
        src="https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg"
        srcSet={undefined}
      />
    </div>
    <div
      className="MuiListItemText-root-301"
    >
      <span
        className="MuiTypography-root-307 MuiTypography-subheading-314 MuiListItemText-primary-304"
      >
        Liana Crooks
      </span>
      <p
        className="MuiTypography-root-307 MuiTypography-body1-316 MuiTypography-colorTextSecondary-329 MuiListItemText-secondary-305"
      >
        Ledner, Johnson and Predovic
      </p>
    </div>
    <span
      className="MuiTouchRipple-root-331"
    />
  </div>
</div>
`;

exports[`ContactInfo renders all contact info correctly 1`] = `
<div>
  <div
    className="MuiPaper-root-156 MuiPaper-elevation1-159 MuiPaper-rounded-157 MuiCard-root-155 contactInfo-card-153"
  >
    <div
      className="MuiCardContent-root-183"
    >
      <div
        className="MuiGrid-container-184 MuiGrid-spacing-xs-24-207"
      >
        <div
          className="MuiGrid-item-185 MuiGrid-grid-xs-12-223 MuiGrid-grid-sm-6-231"
        >
          <img
            alt="Avatar"
            className="contactInfo-img-154"
            src="https://s3.amazonaws.com/uifaces/faces/twitter/kevinoh/128.jpg"
          />
        </div>
        <div
          className="MuiGrid-item-185 MuiGrid-grid-xs-12-223 MuiGrid-grid-sm-6-231"
        >
          <h2>
            Contact Details
          </h2>
          <hr
            className="MuiDivider-root-280"
          />
          <h3>
            General Information
          </h3>
          <h4>
            Liana Crooks
          </h4>
          <h3>
            Contact Information
          </h3>
          <h4>
            Gerry_Hackett77@gmail.com
          </h4>
          <h4>
            (895) 984-0132
          </h4>
          <h3>
            Job
          </h3>
          <h4>
            Investor Functionality Coordinator
          </h4>
          <h4>
            Ledner, Johnson and Predovic
          </h4>
          <h3>
            Address
          </h3>
          <h4>
            Guinea-Bissau - New Devon
          </h4>
          <h4>
            1520 Zemlak Cove
          </h4>
          <h4>
            42586-7898
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
